File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& interval) {
        vector<vector<int>> ans;
        
        sort(begin(interval), end(interval));
        
        int n = interval.size(), q = 0;
        ans.push_back(interval[0]);
        for(int i=1; i<n; i++){
            if(interval[i][0]<=ans[q][1]){
                ans[q][0] = min(ans[q][0], interval[i][0]);
                ans[q][1] = max(ans[q][1], interval[i][1]);
            } else{
                ans.push_back(interval[i]);
                q++;
            }
        }
        
        return ans;
    }
    
};

File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> (a[0] - b[0]));

        int n = intervals.length;

        List<int[]> list = new ArrayList<>();

        list.add(intervals[0]);

        for (int i = 1; i < n; i++) {
            int[] last = list.get(list.size() - 1);
            if (last[1] >= intervals[i][0]) {
                last[1] = Math.max(last[1], intervals[i][1]);
            } else {
                list.add(intervals[i]);
            }
        }

        return list.toArray(new int[list.size()][]);
    }
}